proxmox/nodes/vms/custom_virtiofs_share.go

```diff
@@ -0,0 +1,134 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ */
+
+package vms
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/url"
+	"strings"
+
+	"github.com/bpg/terraform-provider-proxmox/proxmox/types"
+)
+
+// CustomVirtiofsShare handles Virtiofs directory shares.
+type CustomVirtiofsShare struct {
+	DirId       *string           `json:"dirid"                  url:"dirid"`
+	Cache       *string           `json:"cache,omitempty"        url:"cache,omitempty"`
+	DirectIo    *types.CustomBool `json:"direct-io,omitempty"    url:"direct-io,omitempty,int"`
+	ExposeAcl   *types.CustomBool `json:"expose-acl,omitempty"   url:"expose-acl,omitempty,int"`
+	ExposeXattr *types.CustomBool `json:"expose-xattr,omitempty" url:"expose-xattr,omitempty,int"`
+}
+
+// CustomVirtiofsShares handles Virtiofs directory shares.
+type CustomVirtiofsShares map[string]*CustomVirtiofsShare
+
+// EncodeValues converts a CustomVirtiofsShare struct to a URL value.
+func (r *CustomVirtiofsShare) EncodeValues(key string, v *url.Values) error {
+	if r.DirId == nil {
+		return fmt.Errorf("dir_id must be set")
+	}
+
+	if r.ExposeAcl != nil && *r.ExposeAcl && r.ExposeXattr != nil && !*r.ExposeXattr {
```

Comment:
It would be nice to have this constraint defined in the schema, but I don't this this is possible in SDK implementation if there is more that 1 item in the `virtiofs` list :(